pinc/gradual.inc

```diff
@@ -227,3 +228,144 @@ function maybe_output_new_proofer_project_message($project)
         echo "</div>";
     }
 }
+
+/*
+ * Output a callout and encourage the user to work in the round with the
+ * highest backlog that they can work in.
+ */
+function encourage_highest_round($username, $round_id = null)
+{
+    global $code_url, $Round_for_round_id_, $ELR_round, $ACCESS_CRITERIA;
+
+    // This URL fragment gets used a lot in this file
+    $round_url = "$code_url/tools/proofers/round.php?round_id=";
+
+    // get the backlog stats for all rounds sorted largest first
+    $backlog_stats = get_round_backlog_stats(array_keys($Round_for_round_id_));
+    arsort($backlog_stats);
+
+    // get all the rounds a user can work
+    $rounds_can_work_in = array_keys(get_stages_user_can_work_in($username));
+
+    // loop through the needs, highest to lowest, and point them to the
+    // one where they can make the most impact with their current access
+    foreach (array_keys($backlog_stats) as $round_target) {
+        $round = get_Round_for_round_id($round_target);
+
+        // don't encourage people towards $ELR_round, as that isn't really
+        // helpful for most sites
+        if ($round->id == $ELR_round->id) {
```

Comment:
@srjfoo this line should have prevented the code from ever referring you to P1 (the entry level round) regardless if that's where the biggest backlog was or not. Which user were you testing with and I'll see if I can piggy-back on that account.

Reply:
Testing user crickets

Reply:
Ah - is this this dialog?
![Screen Shot 2023-12-24 at 8 41 36 AM](https://github.com/DistributedProofreaders/dproofreaders/assets/5853628/a083f94f-2cab-4730-80a4-c5e0a3bb2ce2)

The Welcome New Proofreader dialog always refers new users back to P1. I think in practice there's no way for both of those dialogs to show up at the same time, but we should make sure of that in code -- assuming this is what you are referring to.

Reply:
We have tons of code in `activity_hub.php`, `gradual.inc`, and `round.php` that uses the number of pages in P1 (and only P1) to show or reveal content and this falls under that category. Somewhat separately, the "proof in higher round" callout in this PR and that code should not be shown together, but they operate on different sources.

I will noodle over how to make this less fragile and hopefully more preditable.